### PR TITLE
Adustments to Mass Plotting Workflow

### PR DESCRIPTION
**Pull request by srivarra**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Addresses a few incorrect and missing features in the mass plotting notebook.

- [x] Adds `dpi` and `fig_size` to both `cohort_cluster_plot` and `color_segmentation_by_stat`
- [x] Fixes incorrect sorting of colors when the colormap file is provided in `cohort_cluster_plot`
- [x] Adds interpolation `"none"` to `color_segmentation_by_stat` to prevent interpolation of the cell borders
- [x] Incorrect colors are properly reported back to the user
- [x] Adds an option to save the plot as any other format accepted by `matplotlib`

**How did you implement your changes**

Added those small changes. The reason the colors were not mapping to the cluster appropriately was because the NumPy Array conversion to the colormaps didn't take into account that the colors need to be sorted w.r.t their associated cluster id.

**Remaining issues**

N/A.

### CODE REVIEW

File: src/ark/utils/plot_utils.py [hunk 1]
```diff
@@ -987,15 +993,17 @@ def cohort_cluster_plot(
...
         unique_clusters: pd.DataFrame = cmd.mapping[[cmd.cluster_column,
                                                      cmd.cluster_id_column]].drop_duplicates()
-        merged_cmap: pd.DataFrame = cmap.merge(right=unique_clusters, on=cmd.cluster_column)
-        cmap_colors = merged_cmap["color"].values
-        colors_like: list[bool] = list(map(colors.is_color_like, cmap_colors))
+        cmap_colors: pd.DataFrame = cmap.merge(
+            right=unique_clusters,
+            on=cmd.cluster_column
+        ).sort_values(by="cluster_id")["color"].values
+        colors_like: list[bool] = [colors.is_color_like(c) for c in cmap_colors]
```
Comment (camisowers) [inline src/ark/utils/plot_utils.py:1000]: What does `is_color_like` do here?

**Review by camisowers — COMMENTED** [review]
Looks good to me! Just one clarifying question.

File: src/ark/utils/plot_utils.py (hunk 1 above)
Comment (srivarra) [reply src/ark/utils/plot_utils.py:1000]: It checks to see if an object is a valid color. For example, it'll work with hex, rgb, rgba, named colors like "orange" and "crimson"

Comment (camisowers) [reply src/ark/utils/plot_utils.py:1000]: gotcha

**Review by camisowers — APPROVED** [review]

**Review by alex-l-kong — APPROVED** [review]
Looks good

**Review by ngreenwald — APPROVED** [review]
Great!

**Pull request merged**